05/DB.test.js

```diff
@@ -1 +1,140 @@
-import DB from './DB';
\ No newline at end of file
+import DB from './DB';
+
+describe('insert', () => {
+    // scenario
+    it('return data when ID is a number', () => {
+        // given
+        const db = new DB();
+        // when
+        const data = { a: 1, b: 2 };
+        const promise = db.insert(data);
+        //then
+        return promise.then(result => {
+            expect(result).toBe(data);
+        });
+    });
+
+
+    it('return ID can be only number', () => {
+        // given
+        const db = new DB();
+        // when
+        const data = { id: 'tak', a: 1, b: 2 };
+        const promise = db.insert(data);
+        //then
+        return promise.catch(err => {
+            expect(err).toBe('ID can be only number!');
+        });
+    });
+
+
+    it('return ID can\'t be duplicated', () => {
+        // given
+        const db = new DB();
+        // when
+        const data1 = { id: 1, a: 1, b: 2 };
+        const data2 = { id: 1, a: 11, b: 22 };
+        db.insert(data1);
+        const promise = db.insert(data2);
```

Comment:
Masz tutaj dwie czynności asynchroniczne więc nie masz pewności, która pierwsza się wykona.
Powinieneś użyć `then` lub `await`, aby poczekać na wykonanie jednej i wykonać kolejną.
Wtedy będziesz miał pewność, że to działa tak jak należy.